model/src/test/test_infection_seeding_method.py

```diff
@@ -0,0 +1,79 @@
+# numpydoc ignore=GL08
+import numpy as np
+import numpy.testing as testing
+from pyrenew.deterministic import DeterministicVariable
+from pyrenew.latent import (
+    SeedInfectionsExponential,
+    SeedInfectionsFromVec,
+    SeedInfectionsZeroPad,
+)
+
+
+def test_seed_infections_exponential():
+    """Check that the SeedInfectionsExponential class generates the correct number of infections at each time point."""
+    n_timepoints = 10
+    rate_RV = DeterministicVariable(0.5, name="rate_RV")
+    I_pre_seed_RV = DeterministicVariable(10.0, name="I_pre_seed_RV")
+    default_t_pre_seed = n_timepoints - 1
+
+    (I_pre_seed,) = I_pre_seed_RV.sample()
+    (rate,) = rate_RV.sample()
+
+    infections_default_t_pre_seed = SeedInfectionsExponential(
+        n_timepoints, rate=rate_RV
+    ).seed_infections(I_pre_seed)
+    infections_default_t_pre_seed_manual = I_pre_seed * np.exp(
+        rate * (np.arange(n_timepoints) - default_t_pre_seed)
+    )
+
+    testing.assert_array_almost_equal(
+        infections_default_t_pre_seed, infections_default_t_pre_seed_manual
+    )
+
+    # assert that infections at default t_pre_seed is I_pre_seed
+    assert infections_default_t_pre_seed[default_t_pre_seed] == I_pre_seed
+
+    t_pre_seed = 6
+    infections_custom_t_pre_seed = SeedInfectionsExponential(
+        n_timepoints, rate=rate_RV, t_pre_seed=t_pre_seed
+    ).seed_infections(I_pre_seed)
+    infections_custom_t_pre_seed_manual = I_pre_seed * np.exp(
+        rate * (np.arange(n_timepoints) - t_pre_seed)
+    )
+    testing.assert_array_almost_equal(
+        infections_custom_t_pre_seed,
+        infections_custom_t_pre_seed_manual,
+        decimal=5,
+    )
+
+    assert infections_custom_t_pre_seed[t_pre_seed] == I_pre_seed
+    return None
+
+
+def test_seed_infections_zero_pad():
```

Comment:
add a case here where we zero-pad a non-scalar `I_pre_seed_RV`?

Reply:
addressed in https://github.com/CDCgov/multisignal-epi-inference/pull/124/commits/20e7b6748948c618f1a1162cea95cfb33e3cb400